File: backend/provisioner/scaling/testdata/go/proxy/proxy.go
package proxy

import (
	"context"

	"ftl/echo"
)

// Proxy returns a greeting
//
//ftl:verb export
func Proxy(ctx context.Context, req string, echo echo.EchoClient) (string, error) {
	return echo(ctx, req)
}

//ftl:verb export
func ProxyCallerIdentity(ctx context.Context, client echo.CallerIdentityClient) (string, error) {
	return client(ctx)
}

File: backend/runner/proxy/testdata/kotlin/proxy/src/main/kotlin/xyz/block/ftl/java/test/proxy/Proxy.kt
package xyz.block.ftl.java.test.proxy

import io.quarkus.logging.Log
import xyz.block.ftl.*
import java.time.ZonedDateTime
import ftl.echo.EchoClient


@Verb
fun proxy(message: String, client: EchoClient): String {
  return client.echo(message)

}
